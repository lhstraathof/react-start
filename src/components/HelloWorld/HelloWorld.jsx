import React, { PureComponent } from 'react';
import { string } from 'prop-types';
import './HelloWorld.scss';

class HelloWorld extends PureComponent {

  static defaultProps = {
    text: 'World',
  };

  static propTypes = {
    /** This should be the description */
    text: string.isRequired,
  };

  render() {
    return (
      <h1 className="h1">Hello {this.props.text}!</h1>
    );
  }
}

export default HelloWorld;