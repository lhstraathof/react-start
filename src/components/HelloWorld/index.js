export { default } from './HelloWorld.jsx';