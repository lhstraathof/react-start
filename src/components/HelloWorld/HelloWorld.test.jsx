import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HelloWorld from './HelloWorld';

configure({ adapter: new Adapter() });

describe('<HelloWorld />', () => {
  it('should correctly render the component', () => {
    const wrapper = shallow(<HelloWorld />);
    expect(wrapper.length).toBe(1);
    expect(wrapper.is('h1')).toBeTruthy();
  });
});

describe('<HelloWorld />', () => {
  const props = {
    text: 'World',
  };

  it('renders correctly with text Hello World!', () => {
    const wrapper = shallow(<HelloWorld {...props} />);
    expect(wrapper.text()).toBe('Hello ' + props.text + '!' );
  });
});

describe('Snapshots', () => {
  it('component matches snapshot', () => {
    const tree = renderer.create(<HelloWorld />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
