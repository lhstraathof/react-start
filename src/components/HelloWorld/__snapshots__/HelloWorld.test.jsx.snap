// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots component matches snapshot 1`] = `
<h1
  className="h1"
>
  Hello 
  World
  !
</h1>
`;
